core/market/decentralized/src/api.rs

```diff
@@ -14,7 +15,7 @@ pub struct PathAgreement {
 
 #[derive(Deserialize)]
 pub struct PathSubscription {
-    pub subscription_id: String,
+    pub subscription_id: SubscriptionId,
```

Comment:
If Subscription id is invalid and can't be parsed to our structure, how will be error handled and what will be response?

Reply:
good catch

Reply:
it needed explicit defining of serde for `SubscriptionId` but now `SubscriptionParseError`'s are handled cleanly at HTTP Request deserialization. No need to expose it from lib.